File: ObjC-Password-Entropy/ObjC-Password-Entropy/L3SDK/Security/L3SDKDefaultCharacterPools.m
//
//  L3SDKDefaultCharacterPools.m
//  ObjC-Password-Entropy
//
//  Created by Domenico Vacchiano on 31/12/15.
//  Copyright © 2015 DomenicoVacchiano. All rights reserved.
//

#import "L3SDKDefaultCharacterPools.h"

@interface L3SDKDefaultCharacterPools ()
@property (nonatomic,strong)NSArray*defaultCharacterSet;
@end

//it represents the standard pool of characters used to evaluate the password entropy
@implementation L3SDKDefaultCharacterPools

-(instancetype) init{
    
    self=[super init];
    
    if (self) {
        self.defaultCharacterSet=[[NSArray alloc]initWithObjects:
                                  //lowercase letters
                                  @"abcdefghijklmnopqrstuvwxyz",
                                  //uppercase letters
                                  @"ABCDEFGHIJKLMNOPQRSTUVWXYZ",
                                  //numbers
                                  @"0123456789",
                                  //symbols above numbers
                                  @"!@#$%^&*()",
                                  //other symbols
                                  @"`~-_=+[{]}\\|;:'\",<.>/?",
                                  //space
                                  @" ",
                                  nil];
    }
    return self;
}



-(NSUInteger) count {
    return [self.defaultCharacterSet count];
}
-(id)objectAtIndex:(NSUInteger)index{
    NSAssert(index < self.count, @"The index is out of bounds");
    return [self.defaultCharacterSet objectAtIndex:index];
}
@end

File: Swift-Password-Entropy/Swift-Password-Entropy/L3SDK/Security/L3SDKDefaultCharacterPools.swift
//
//  L3SDKDefaultCharacterPools.swift
//  Swift-Password-Entropy
//
//  Created by Domenico Vacchiano on 31/12/15.
//  Copyright © 2015 DomenicoVacchiano. All rights reserved.
//

import UIKit

class L3SDKDefaultCharacterPools: NSArray {
    
    //it represents the standard pool of characters used to evaluate the password entropy
    
    let defaultCharacterSet =  [
        //lowercase letters
        "abcdefghijklmnopqrstuvwxyz",
        //uppercase letters
        "ABCDEFGHIJKLMNOPQRSTUVWXYZ",
        //numbers
        "0123456789",
        //symbols above numbers
        "!@#$%^&*()",
        //other symbols
        "`~-_=+[{]}\\|;:'\",<.>/?",
        //space
        " "
    ]
    
    override var count: Int {
        get {
            return defaultCharacterSet.count
        }
    }
    
    override func objectAtIndex(index: Int) -> AnyObject {
        assert(index < count, "The index is out of bounds")
        return defaultCharacterSet[index]
    }
    
}

